### ISSUE TITLE
Clarify didChangeWatchedFiles 'non dynamic' meaning

### ISSUE THREAD

**Issue by kdvolder** [Microsoft/language-server-protocol#615]
I'm trying to understand the intention of the spec around 'didChangeWatchedFiles'. I think I understand the behavior for when 'dynamicRegistration' flag is set to  'true' in ClientCapabilities. In a nutshel, in that case, the server can register dynamically with specific glob patterns the server is interested in.

However, if 'dynamicRegistration' is set to false. What does that mean then? 

Following along with other 'static' regustrations, I was expecting there would be a something in ServerCapabilities (returned by server initialization) where a server could somehow declare the patterns of files its interested in. But there doesn't seem to be anything like that.

So that brings me to my question, how does a server, in the static case, declare which file changes the server is interested in?

**Comment by dbaeumer** [comment]
Good point. You can't right now. In a first implementation I made that a configuration in the client library. 

But it would be quite straight forward to add it. Code wise here: https://github.com/Microsoft/vscode-languageserver-node/blob/dbaeumer/declarationSupport/client/src/client.ts#L1148

The `DidChangeWatchedFilesRegistrationOptions` can be reused for the server caoabilities.

Interested in providing a PR?

**Comment by dbaeumer** [comment]
I clarified this as suggested. Still keeping this open to add static file watching configuration.

**Comment by radeksimko** [comment]
It makes sense for "generic" (not language-specific) clients to need it in order to decide which server to talk to when opening files/folders - I assume this is also essential for multiple servers with overlapping functionality co-existing.

IMO though the server typically holds the most knowledge about the language and should know better what extensions are actually relevant also know this may differ between language versions.

So I'm thinking if the server is allowed to "announce" its own watchers, the client should always honour these and only have the option to reduce the list, but _not_ expand.

i.e. it wouldn't make much sense for a client to send `*.py` changes to a server which only supports `*.go`

This assumes clients implement the filtering correctly and run paths first through client's and then server's glob patterns, so that e.g. if client registers for `*.*`, server supports `*.go` and `main.go` changes, the notification still goes through.
